run_wml_tests

```diff
@@ -153,24 +177,37 @@ class WesnothRunner:
             print(repr(args))
         try:
             res = run_with_rerun_for_sdl_video(args, timeout)
-        except subprocess.TimeoutExpired as e:
+        except subprocess.TimeoutExpired as t:
             print("Timed out (killed by Python timeout implementation)")
-            res = subprocess.CompletedProcess(args, UnitTestResult.TIMEOUT.value)
-        if self.verbose > 1:
-            print("Result:", res.returncode)
+            res = subprocess.CompletedProcess(args, UnitTestResult.TIMEOUT.value, t.output)
+        if self.verbose > 0:
+            print(res.stdout.decode('utf-8'))
+            if self.verbose > 1:
+                print("Result:", res.returncode)
+        returned_result = UnitTestResult(res.returncode)
+        num_passed = 0
+        if test_list[0].status == UnitTestResult.PASS:
+            num_passed = res.stdout.count(b"PASS TEST")
+            test_summary.pass_test(num_passed)
+        elif returned_result == expected_result:
+            num_passed = 1
+            test_summary.pass_test()
         if res.returncode < 0:
             print("Wesnoth exited because of signal", -res.returncode)
             if options.backtrace:
                 print("Launching GDB for a backtrace...")
                 gdb_args = ["gdb", "-q", "-batch", "-ex", "start", "-ex", "continue", "-ex", "bt", "-ex", "quit", "--args"]
                 gdb_args.extend(args)
                 subprocess.run(gdb_args, timeout=240)
+            test_summary.crash_test()
+            test_summary.skip_test(len(test_list) - num_passed - 1)
             raise UnexpectedTestStatusException()
-        if res.returncode != expected_result.value:
-            with open(get_output_filename(args), "r") as output:
-                for line in output.readlines():
-                	print(line)
-            print("Failure, Wesnoth returned", res.returncode, "but we expected", expected_result.value)
+        if returned_result != expected_result:
+            if self.verbose == 0:
+                print(res.stdout.decode('utf-8'))
```

Comment:
This was already printed on line 186.

Reply:
Your line number is off, but... the first print is only if -v is passed, this one is if -v was not passed.

Reply:
In other words, without -v it only prints the full output when the test fails; with -v it _always_ prints the full output.